cluster-autoscaler/core/podlistprocessor/pod_list_processor.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package podlistprocessor
+
+import (
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/autoscaler/cluster-autoscaler/context"
+)
+
+type defaultPodListProcessor struct {
+	currentlyDrainedNodes *currentlyDrainedNodesPodListProcessor
+	filterOutSchedulable  *filterOutSchedulablePodListProcessor
+}
+
+// NewDefaultPodListProcessor returns a default implementation of the pod list
+// processor, which wraps and sequentially runs other sub-processors.
+func NewDefaultPodListProcessor(currentlyDrainedNodes *currentlyDrainedNodesPodListProcessor, filterOutSchedulable *filterOutSchedulablePodListProcessor) *defaultPodListProcessor {
```

Comment:
I'd prefer having a generic pod list processor aggregating other pod list processors, but I remember having a similar discussion in the past with opinions on both sides and I'm not sure if we agreed on anything back then. I guess either one is fine then.

Reply:
I'm not convinced by this - the concept of order is definitely there in a generic aggregator, I don't see why determining it during initialization would be less safe. But then again, not going to block on this. :)